src/model/test/Space_GTest.cpp

```diff
@@ -3220,10 +3230,508 @@ TEST_F(ModelFixture, Space_Polyhedron_Volume) {
+    ASSERT_TRUE(s_);
+    auto [x, y] = getPos(counter++);
+    s_->setXOrigin(x);
+    s_->setYOrigin(y);
+    EXPECT_TRUE(s_->isConvex());
+    EXPECT_TRUE(s_->findNonConvexSurfaces().empty());
+  }
+
+  {
+    std::vector<Point3d> points = hShapedPolygon(p0, total_length);
+    std::reverse(points.begin(), points.end());
+    auto s_ = Space::fromFloorPrint(points, 2, m, "hShapedPolygon");
+    ASSERT_TRUE(s_);
+    auto [x, y] = getPos(counter++);
+    s_->setXOrigin(x);
+    s_->setYOrigin(y);
+    EXPECT_FALSE(s_->isConvex());
+    // Roof and floor
+    EXPECT_EQ(2, s_->findNonConvexSurfaces().size());
+    EXPECT_EQ("hShapedPolygon Floor", s_->findNonConvexSurfaces().front().nameString());
+    EXPECT_EQ("hShapedPolygon RoofCeiling", s_->findNonConvexSurfaces().back().nameString());
+  }
+
+  {
+    std::vector<Point3d> points = hShapedPolygon(p0, total_length);
+    std::reverse(points.begin(), points.end());
+    auto s_ = Space::fromFloorPrint(points, 2, m, "hShapedPolygon");
+    ASSERT_TRUE(s_);
+    auto [x, y] = getPos(counter++);
+    s_->setXOrigin(x);
+    s_->setYOrigin(y);
+    EXPECT_FALSE(s_->isConvex());
+    // Roof and floor
+    EXPECT_EQ(2, s_->findNonConvexSurfaces().size());
+  }
+
+  {
+    std::vector<Point3d> points = tShapedPolygon(p0, total_length);
+    std::reverse(points.begin(), points.end());
+    auto s_ = Space::fromFloorPrint(points, 2, m, "tShapedPolygon");
+    ASSERT_TRUE(s_);
+    auto [x, y] = getPos(counter++);
+    s_->setXOrigin(x);
+    s_->setYOrigin(y);
+    EXPECT_FALSE(s_->isConvex());
+    // Roof and floor
+    EXPECT_EQ(2, s_->findNonConvexSurfaces().size());
+  }
+
+  {
+    std::vector<Point3d> points = lShapedPolygon(p0, total_length);
+    std::reverse(points.begin(), points.end());
+    auto s_ = Space::fromFloorPrint(points, 2, m, "lShapedPolygon");
+    ASSERT_TRUE(s_);
+    auto [x, y] = getPos(counter++);
+    s_->setXOrigin(x);
+    s_->setYOrigin(y);
+    EXPECT_FALSE(s_->isConvex());
+    // Roof and floor
+    EXPECT_EQ(2, s_->findNonConvexSurfaces().size());
+  }
+
+  {
+    std::vector<Point3d> points = squaredPolygon(p0, total_length);
+    std::reverse(points.begin(), points.end());
+    auto s_ = Space::fromFloorPrint(points, 2, m, "squaredPolygon");
+    ASSERT_TRUE(s_);
+    auto [x, y] = getPos(counter++);
+    s_->setXOrigin(x);
+    s_->setYOrigin(y);
+    EXPECT_TRUE(s_->isConvex());
+    EXPECT_TRUE(s_->findNonConvexSurfaces().empty());
+  }
+
+  {
+    std::vector<Point3d> points = rectangularPolygon(p0, total_length * 2, total_length);
+    std::reverse(points.begin(), points.end());
+    auto s_ = Space::fromFloorPrint(points, 2, m, "rectangularPolygon");
+    ASSERT_TRUE(s_);
+    auto [x, y] = getPos(counter++);
+    s_->setXOrigin(x);
+    s_->setYOrigin(y);
+    EXPECT_TRUE(s_->isConvex());
+    EXPECT_TRUE(s_->findNonConvexSurfaces().empty());
+  }
+
+  // m.save("Test_convexity.osm", true);
```

Comment:
This tests the StandardShapes.hpp and the Space::isConvex / Space::findNonConvexSurfaces.

Resulting model:

![Regular_shapes](https://user-images.githubusercontent.com/5479063/228518236-7e8730df-1774-48c1-a902-b43000d135f0.png)